### PR TITLE
Clean up several minor issues in setPage

### PR DESCRIPTION
**Pull request by aarongoldenthal**
The following issues were noticed today in the `setPage` function (all are low priority cleanup).

- Removed unused `url` argument from `setPage`
- Moved debug log entry for navigating to URL to proper location (from `setPage` to `gotoUrl`)
- Update debug logging for all `setPage` operations

Debug log changes:

```diff
 > pa11y -d https://pa11y.org

 Welcome to Pa11y

  > Running Pa11y on URL https://pa11y.org
  > Debug: Launching Headless Chrome
- > Debug: Opening URL in Headless Chrome
+ > Debug: Opening new browser page
+ > Debug: Setting page User-Agent to "pa11y/8.0.0"
+ > Debug: Setting page viewport
+ > Debug: Navigating to https://pa11y.org
  > Debug: Loading runner: htmlcs
  > Debug: Injecting Pa11y
  > Debug: Injecting runner: htmlcs
  > Debug: Running Pa11y on the page
  > Debug: Document title: "Pa11y"

 No issues found!
```

### CODE REVIEW

**Review by danyalaytekin — COMMENTED** [review]
(Oops, let me try again using the review button.)

Well spotted @aarongoldenthal!

1. I think you've completed a [reversion which hadn't quite concluded](https://github.com/pa11y/pa11y/commit/52af3fc5e6963c26f575ef104c3b36d5ca8c19fa).  But this happened a while ago so I should check... are you happy with this signature change @josebolos or was it left there pending a restoration of the reverted feature?
2. Checking that commit I also noticed that we have some testing around the debug logging, which it could make sense to expand to cover the changes in this PR, for consistency, if that's our practice.  Alternatively the extra logging could be pushed into a separate PR.  @aarongoldenthal @josebolos

**Comment by josebolos** [comment]
Hi all! Sorry I'm a bit late to the party. This seems absolutely fine to me, I tried to find the logs for the GH actions workflows but it looks like the PR is so old that GH has deleted the logs and I haven't found a way to trigger them again for a PR.

I don't think that anyone would be using the setPage externally (and, if so, what a terrible thing to do) so I think it should probably be fine to release as a bugfix.
